### DISCUSSION TITLE
Input Map settings "Filter by Event" input prevents navigating the UI with the keyboard

### DISCUSSION THREAD

**Discussion by harrison-munitz** [godotengine/godot-proposals#11217] [Editor]
## Problem

In `Project Settings`, under the `Input Map` tab, the second input is a box labeled `Filter by Event` which when focused, records any button presses to be used to filter the input list. 

If you try to navigate through the screen using a keyboard, as soon as you pass that input, it starts capturing all keyboard inputs, recording and listing the `Tab` or Arrow key button presses and preventing you from navigating through the screen with the keyboard anymore.

This is both an accessibility and QOL problem, as it makes it effectively impossible to edit this part of the settings by keyboard only.

## Suggestion

Ideally, you would need to press `Enter` or something to initiate using the box, at least if focused by keyboard and not the mouse, and ideally there would also be some way to escape the keyboard capture using only the keyboard, like holding `Escape` for a second, or something like that.

**Comment by Calinou** [comment]
What I'd do is the following:

- Ignore <kbd>Tab</kbd> and <kbd>Shift + Tab</kbd> in the input filter and editing dialog.
- Make pressing <kbd>Escape</kbd> close the input filter and editing dialog if the current key is Escape already (in other words, press <kbd>Escape</kbd> twice to close the dialog).
- Arrow keys should still be listened to, as they don't move UI focus when on a text field and you can't actually *write* text in those text fields, so there is no text to navigate within.

Additionally, I noticed you can't tab to the Close button of the project settings dialog (nor the Accept/Cancel buttons of the input editing dialog). We should fix this so that dialogs are fully usable on a keyboard again. (It works in Godot 3.5.3 but not 4.3.)
